docs/how_to/apphooks.rst

```diff
@@ -8,288 +10,186 @@ this case, you can create a normal django CMS page without any content of its
 own, and attach the news application to the page; the news application's content
 will be delivered at the page's URL.
 
-To create an apphook place a ``cms_apps.py`` in your application. And in it write
-the following::
+All URLs in that URL path will be passed to the attached application's URL configs.
+
+The :ref:`Tutorials <tutorials>` section contains a basic guide to :ref:`getting started with apphooks
+<apphooks_introduction>`. This document assumes more familiarity with the CMS generally.
+
+******************************
+The basics of apphook creation
+******************************
+
+To create an apphook, create a ``cms_apps.py`` file in your application.
+
+.. note:: Up to version 3.1 this module was named ``cms_app.py`` - please
+          update your existing modules to the new naming convention.
+          Support for the old name will be removed in version 3.4.
+
+The file needs to contain a :class:`CMSApp <cms.app_base.CMSApp>` sub-class, like::
 
     from cms.app_base import CMSApp
     from cms.apphook_pool import apphook_pool
     from django.utils.translation import ugettext_lazy as _
 
     class MyApphook(CMSApp):
         name = _("My Apphook")
-        _urls = ["myapp.urls"]
+        _urls = ["myapp.urls"]       # replace this with the path to your application's URLs module
 
     apphook_pool.register(MyApphook)
 
 .. versionchanged:: 3.3
-    ``CMSApp.urls`` has been replaced by ``CMSApp._urls``; previous attribute is
-    now deprecated and will be removed in version 3.5
-
-.. versionchanged:: 3.3
-    ``CMSApp.menus`` has been replaced by ``CMSApp._menus``; previous attribute is
-    now deprecated and will be removed in version 3.5
+    ``CMSApp._urls`` replaces ``CMSApp.urls``. ``urls`` is now deprecated and will be removed in
```

Comment:
should we really advocate the use of a "private" attribute? I think the public interface is `get_urls`.
/cc @yakky